backend/server.js

```diff
@@ -0,0 +1,31 @@
+const {applyMiddleware} = require("graphql-middleware");
+const {makeExecutableSchema} = require("graphql-tools");
+
+const {ApolloServer} = require('apollo-server');
+const typeDefs = require('./schema.js');
+const resolvers = require('./resolvers.js');
+const permissions = require('./permissions.js');
+const {getContext} = require('./helper/context.js');
+
+const schema = applyMiddleware(
+    makeExecutableSchema({
+        typeDefs,
+        resolvers
+    }),
+    permissions,
+);
+// The ApolloServer constructor requires two parameters: your schema
+// definition and your set of resolvers.
+function getApolloServer() {
+    return new ApolloServer({
+        schema, context: ({req}) => {
```

Comment:
If the method signature of `getContext` would be an object that contains `req` you could refactor this code to:
```suggestion
        schema,
        context: getContext,
```
That's it!